### ISSUE TITLE
Custom kernel, bootargs and flash drives

### ISSUE THREAD

**Issue by edubart** [sunodo/sunodo#193]
## 📚 Context

I am experimenting a dapp where I need a custom kernel, custom bootargs and extra flash drives. Currently there is no way to do this with Sunodo.

## ✔️ Solution

Would be nice if Sunodo could allow me to specify a different kernel image, different kernel headers, extra bootargs and extra flash drive configurations.

A plus would be a label to pass custom arguments to the `cartesi-machine` command, this would allow to workaround some of these limitations without Sunodo having official support yet.

**Comment by tuler** [comment]
Regarding the kernel, it's part of the `sunodo/sdk:0.2.0` image, which is built at https://github.com/sunodo/sunodo/tree/main/packages/sdk

Sunodo `build` looks for that image, which can be customized using the Docker label `io.sunodo.sdk_version`.
If you define in the Dockerfile:

```
LABEL io.sunodo.sdk_version=my-custom-kernel
```

It will use the docker image `sunodo/sdk:my-custom-kernel`.

This could still be improved to replace the entire docker image name, not only the tag.

**Comment by tuler** [comment]
I'm considering to move away from the configuration through Docker labels and introducing a sunodo.yaml configuration file.

Any thoughts?

**Comment by edubart** [comment]
> Regarding the kernel, it's part of the sunodo/sdk:0.2.0

I am doing that way now, it works. But looks like this `LABEL` config does not reflect `sunodo shell` command.

> I'm considering to move away from the configuration through Docker labels and introducing a sunodo.yaml configuration file.

I think it makes sense, this would open the possibility to customize other commands too, not just `sunodo build`. Like it could have also configs for `sunodo run`, and others

**Comment by tuler** [comment]
> But looks like this `LABEL` config does not reflect `sunodo shell` command.

You're right, it's a known issue. See comment https://github.com/sunodo/sunodo/blob/26169e278a3004d6139b82f4aa05560f457ace76/apps/cli/src/commands/shell.ts#L34

**Comment by gbarros** [comment]
I think it makes a lot of sense to have a `sunodo.yaml`; many CLIs have it, and I feel it's a natural progression. 
I would add that support to define env vars is nice to have =]

**Comment by gligneul** [comment]
I'm writing a few examples that use sunodo, but they require a specific sunodo version (e.g., 0.9). If someone tries to run my example with a previous or future version, it will break internally with an error that is hard to understand. To solve this, it would be nice to specify which sunodo version my example uses in the `sunodo.yaml` file. Then, if the user tries to run my example with the wrong version, sunodo will raise a presentable error telling them to use the correct version.

**Comment by edubart** [comment]
Be aware that custom flash drives will be a requirement the for running multiple Hypervisor guests in the future (each flash drive will be a different rootfs).
